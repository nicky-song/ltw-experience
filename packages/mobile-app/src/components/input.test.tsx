import Input from './input';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '../test/testing';

describe('Components Input', () => {
  it('should match snapshot', () => {
    const handleChange = jest.fn();
    const input = renderer.create(
      <Input
        iconName='email-outline'
        placeholder='Email'
        onChangeText={handleChange}
      />,
    );

    expect(input).toMatchSnapshot();
  });
  it('should call onFocus', () => {
    const onFocus = jest.fn();
    render(
      <Input iconName='lock' testID='password-input' onChangeText={onFocus} />,
    );
    const passwordInput = screen.getByTestId('password-input');

    fireEvent.press(passwordInput);
    fireEvent.changeText(passwordInput, 'password');

    expect(onFocus).toHaveBeenCalled();
  });
});
