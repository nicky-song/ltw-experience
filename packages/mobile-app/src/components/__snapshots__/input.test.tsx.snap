// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Input should match snapshot 1`] = `
<View
  style={
    {
      "marginBottom": 24,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "grey",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "paddingHorizontal": 10,
          "width": "100%",
        },
        null,
      ]
    }
  >
    <Text />
    <TextInput
      onChangeText={[MockFunction]}
      placeholder="Email"
      style={
        {
          "flex": 1,
          "height": "100%",
        }
      }
    />
  </View>
</View>
`;
